google/cloud/security/inventory/pipelines/load_groups_pipeline.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pipeline to load GSuite Account Groups into Inventory."""
+
+from oauth2client.contrib.gce import AppAssertionCredentials
+from oauth2client.service_account import ServiceAccountCredentials
+
+from google.cloud.security.common.data_access.errors import CSVFileError
+from google.cloud.security.common.data_access.errors import MySQLError
+from google.cloud.security.common.gcp_api._base_client import ApiExecutionError
+# TODO: Investigate improving so the pylint disable isn't needed.
+# pylint: disable=line-too-long
+from google.cloud.security.common.gcp_api.admin_directory import AdminDirectoryClient
+from google.cloud.security.common.util import metadata_server
+from google.cloud.security.common.util.log_util import LogUtil
+from google.cloud.security.inventory import transform_util
+from google.cloud.security.inventory.errors import LoadDataPipelineError
+
+
+LOGGER = LogUtil.setup_logging(__name__)
+RESOURCE_NAME = 'groups'
+REQUIRED_SCOPES = ['https://www.googleapis.com/auth/admin.directory.user',
+                   'https://www.googleapis.com/auth/admin.directory.group']
+
+
+def _can_inventory_google_groups(config):
+    """A simple function that validates required inputs for inventorying groups.
+
+    Args:
+        config: Dictionary of configurations built from our config.
+
+    Returns:
+        Boolean
+    """
+    required_gcp_execution_config = [config.get('service_account_email'),
+                                     config.get('domain_super_admin_email')]
+
+    required_local_execution_config = [
+        config.get('service_account_email'),
+        config.get('service_account_credentials_file'),
+        config.get('domain_super_admin_email')]
+
+    if metadata_server.can_reach_metadata_server():
```

Comment:
nit: Is there any reason why both the local and gcp execution config needs to be checked?  Maybe we can determine the environment separately, and check the execution config once?

Do you think this might be better and easier to read??

```python
    if is_gce():
        required_execution_config = required_gcp_execution_config
    else:
        required_execution_config = required_local_execution_config

    if metadata_server.can_reach_metadata_server():
        if False in required_execution_config:
            return False
        return True
```